Tooling/ScriptGenerator/ExtensionMethods.cs

```diff
@@ -0,0 +1,178 @@
+    /// </summary>
+    /// <param name="value">This <c>string</c> to search for the <paramref name="beforeChar"/> and <paramref name="afterChar"/>.</param>
+    /// <param name="beforeChar">The <c>char</c> that might be located before the <paramref name="afterChar"/>.</param>
+    /// <param name="afterChar">The <c>char</c> that might be located after the <paramref name="beforeChar"/>.</param>
+    /// <returns><c>true</c> if the <paramref name="beforeChar"/> is located before the <paramref name="afterChar"/>.</returns>
+    public static bool IsBefore(this string value, char beforeChar, char afterChar)
+    {
+        if (string.IsNullOrEmpty(value))
+        {
+            return false;
+        }
+
+        var beforeCharIndex = value.IndexOf(beforeChar);
+        var afterCharIndex = value.IndexOf(afterChar);
+
+        if (beforeCharIndex == -1)
+        {
+            return false;
+        }
+
+        if (afterCharIndex == -1)
+        {
+            return true;
+        }
+
+        return beforeCharIndex < afterCharIndex;
+    }
+
+    /// <summary>
+    /// Returns a value indicating whether or not the first occurence of the given <paramref name="beforeStr"/>
+    /// is located positionally before the given <paramref name="afterChar"/>.
+    /// </summary>
+    /// <param name="value">This <c>string</c> to search for the <paramref name="beforeStr"/> and <paramref name="afterChar"/>.</param>
+    /// <param name="beforeStr">The <c>char</c> that might be located before the <paramref name="afterChar"/>.</param>
+    /// <param name="afterChar">The <c>char</c> that might be located after the <paramref name="beforeStr"/>.</param>
+    /// <returns><c>true</c> if the <paramref name="beforeStr"/> is located before the <paramref name="afterChar"/>.</returns>
+    public static bool IsBefore(this string value, string beforeStr, char afterChar)
+    {
+        if (string.IsNullOrEmpty(value))
+        {
+            return false;
+        }
+
+        var beforeStrIndex = value.IndexOf(beforeStr, StringComparison.Ordinal);
+
+        if (beforeStrIndex == -1)
+        {
+            return false;
+        }
+
+        var afterCharIndex = value.IndexOf(afterChar);
+
+        if (afterCharIndex == -1)
+        {
+            return true;
+        }
+
+        return beforeStrIndex < afterCharIndex;
+    }
+
+    /// <summary>
+    /// Returns a value indicating whether or not the first occurence of the given <paramref name="afterChar"/>
+    /// is located positionally after the given <paramref name="beforeChar"/>.
+    /// </summary>
+    /// <param name="value">This <c>string</c> to search for the <paramref name="afterChar"/> and <paramref name="beforeChar"/>.</param>
+    /// <param name="afterChar">The <c>char</c> that might be located after the <paramref name="beforeChar"/>.</param>
+    /// <param name="beforeChar">The <c>char</c> that might be located before the <paramref name="afterChar"/>.</param>
```

Comment:
I don't like "might".   You can also use just "may be located...", or "can be located..."
```suggestion
    /// <param name="beforeChar">The <c>char</c> that may or may not be located before the <paramref name="afterChar"/>.</param>
    /// <param name="afterChar">The <c>char</c> that may or may not be located after the <paramref name="beforeChar"/>.</param>
```